app/controllers/index.js

```diff
@@ -37,42 +53,85 @@ export default Ember.Controller.extend({
   },
 
   actions: {
-    handleClick(event) {
-      const selection = this.get('selection');
-      const summaryLevel = selection.summaryLevel;
+    handleClick(e) {
+      if (!this.get('drawMode')) {
+        const selection = this.get('selection');
+        const summaryLevel = selection.summaryLevel;
+
+        let layers = [];
+
+        switch (summaryLevel) { // eslint-disable-line
+          case 'tracts':
+            layers = ['census-tracts-fill'];
+            break;
+          case 'blocks':
+            layers = ['census-blocks-fill'];
+            break;
+          case 'ntas':
+            layers = ['neighborhood-tabulation-areas-fill'];
+            break;
+          case 'pumas':
+            layers = ['nyc-pumas-fill'];
+            break;
+        }
+
+        const [found] =
+          e.target.queryRenderedFeatures(
+            e.point,
+            { layers },
+          );
+
+
+        if (found) {
+          selection.handleSelectedFeatures([found]);
+        }
+      }
+    },
 
-      let layers = [];
-
-      switch (summaryLevel) { // eslint-disable-line
-        case 'tracts':
-          layers = ['census-tracts-fill'];
-          break;
-        case 'blocks':
-          layers = ['census-blocks-fill'];
-          break;
-        case 'ntas':
-          layers = ['neighborhood-tabulation-areas-fill'];
-          break;
-        case 'pumas':
-          layers = ['nyc-pumas-fill'];
-          break;
+    handleDrawButtonClick() {
+      const drawMode = this.get('drawMode');
+      if (drawMode) {
+        draw.trash();
+        this.set('drawMode', false);
+      } else {
+        draw.changeMode('draw_polygon');
+        this.set('drawMode', true);
       }
+    },
 
-      const [found] =
-        event.target.queryRenderedFeatures(
-          event.point,
-          { layers },
-        );
+    handleDrawCreate(e) {
+      // delete the drawn geometry
+      draw.deleteAll();
 
+      const selection = this.get('selection');
+      const summaryLevel = selection.summaryLevel;
 
-      if (found) {
-        selection.handleSelectedFeature([found]);
-      }
+      const geometry = e.features[0].geometry;
+      geometry.crs = {
+        type: 'name',
+        properties: {
+          name: 'EPSG:4326',
+        },
+      };
+
+      const intersectionSQL = generateIntersectionSQL(summaryLevel, geometry);
+      carto.SQL(intersectionSQL, 'geojson', 'post')
+        .then((FC) => {
```

Comment:
best to spell out `featureCollection` or name what feature collection represents (`underlyingFeatures`)